docs/arcadia-users-group/20221011-developer-productivity/lesson.md

```diff
@@ -0,0 +1,172 @@
+
+Once you select the color scheme you'd like to use, just do the following:
+
+- Launch iTerm2
+- Type CMD+i
+- Navigate to Colors tab
+- Click on Load Presets
+- Click on Import
+- Select your file
+- Click on Load Presets and choose a color scheme
+
+### Tweaking bash
+
+In general, most customizations go to the `.bash_profile` or `.bashrc` files. Here, we'll show a slightly modular approach, so we can simplify the files before they get too long and difficult to manage. Once you change the `.bash_profile` or the underlying configuration files, you need to make sure those changes are persisted. You can that by "sourcing" the `.bash_profile` with `source .bash_profile` or restarting your terminal.
+
+#### Tweaking the bash prompt
+
+For this and the following sections, we need to create the following files: `.bash_profile` (main entry point the Terminal application searches for), `.bash_prompt` (customizes the look of the prompt), `.aliases` (adds new shortcuts/aliases for common commands) and `.gitconfig` (customizations for git commands). Let's start with creating these:
+
+```{bash}
+cd ~
+touch .bash_profile .bash_prompt .aliases .gitconfig
+code .bash_profile .bash_prompt .aliases .gitconfig
+```
+
+These commands will not recreate the files if they already exist. If they already exist, we'll edit them. My versions of all these files can be found in the `config-and-dotfiles` directory. You can just copy and paste them, making sure to not delete anything in your existing files. We'll go through them one by one, starting with the [.bash_prompt](config-and-dotfiles/.bash_prompt).
+
+This file basically shapes the main prompt that shows up when you use bash. For it to be activated, make sure you also copy the [.bash_profile](config-and-dotfiles/.bash_profile) file. And run `source ~/.bash_profile`.
+
+#### Using aliases for commonly used commands
+
+Aliases are shortcuts for commands that you frequently use. Some of these provide stylistic changes. Some of these provide useful shortcuts for common git commands. You can customize this however you want, but a simplified version of my aliases can be found [here](config-and-dotfiles/.aliases).
+
+#### Tweaking git
+
+Git uses the `.gitconfig` file to configure the behavior of the git CLI. My simplified version of the configuration file can be found [here](config-and-dotfiles/.gitconfig). The two big components of this configuration file are:
+
+- Colorful view for git branches, stages etc.
+- Better visualization of the git history through the command-line
+
+## What will happen in the future?
+
+- We got a glimpse of basic forms of linting with Pylance and basic forms of formatting with Prettier. We'll be doing a more in-depth workshop on Python and R specific linters (and ways of automating the linting process) so you can easily catch errors and have consistent styling for your code
```

Comment:
should we plan this for after the conda stuff?

Reply:
Sure!